### PR TITLE
[WIP] Enable SASL, DKIM

### PR DESCRIPTION
**Pull request by timothy-spencer**
This enables SASL and DKIM support for the release.  It also updates some docs and some other random stuff.

This is a WIP right now because I am unsure why postfix/opendkim don't restart when I do a deploy.  Otherwise, it all seems to work.

### CODE REVIEW

File: jobs/postfix/templates/bin/ctl [hunk 1]
```diff
@@ -11,12 +11,11 @@ case $1 in
 
   start)
       /etc/init.d/postfix start
-
       ;;
 
   stop)
       /etc/init.d/postfix stop
-
+      sleep 5
```
Comment (LinuxBozo) [inline jobs/postfix/templates/bin/ctl:18]: Why is the sleep necessary in a control script?

File: jobs/postfix/spec [hunk 2]
```diff
@@ -43,3 +51,14 @@ properties:
     tls_key:
       description: "TLS Key"
       default: ""
+    sasl_users:
+      description: "list of users and passwords used for SASL auth"
+      default:
```
Comment (LinuxBozo) [inline jobs/postfix/spec:56]: Shouldn't have `default` users. Use an `example` instead.

File: jobs/postfix/templates/opendkim/etc/opendkim/domainkeys/cloudgov.private.erb [hunk 3]
```diff
@@ -0,0 +1 @@
+<% if_p('postfix.dkim_privatekey') do %><%= p('postfix.dkim_privatekey') %><% end %>
```
Comment (LinuxBozo) [inline jobs/postfix/templates/opendkim/etc/opendkim/domainkeys/cloudgov.private.erb:1]: I get that we're probably stuck maintaining this for a while, but `cloudgov.private.erb`?  Probably should rethink this filename so this package is re-usable.

**Review by sharms — APPROVED** [review]

File: jobs/postfix/templates/bin/ctl (hunk 1 above)
Comment (timothy-spencer) [reply jobs/postfix/templates/bin/ctl:18]: it was left over from a test.  I will remove it!

File: jobs/postfix/spec (hunk 2 above)
Comment (timothy-spencer) [reply jobs/postfix/spec:56]: Ah.  Did not know that was a possibility.  Will fix that.

File: jobs/postfix/templates/opendkim/etc/opendkim/domainkeys/cloudgov.private.erb (hunk 3 above)
Comment (timothy-spencer) [reply jobs/postfix/templates/opendkim/etc/opendkim/domainkeys/cloudgov.private.erb:1]: Good point.  Will fix the name.

**Pull request merged**